luigi/worker.py

```diff
@@ -534,8 +531,22 @@ def add(self, task, multiprocess=False):
             task.trigger_event(Event.BROKEN_TASK, task, ex)
             self._email_unexpected_error(task, formatted_traceback)
         finally:
-            pool.close()
-            pool.join()
+            # We must not "pool.join()" here, as the pool proccesses may have
+            # deadlocked (since this main tread is not .get()-ing from the
+            # queue anymore).
+            # http://stackoverflow.com/a/27492515/621449
+            try:
+                # However, as some users have disabled garbage collection (see
```

Comment:
this is technically not correct, we want to terminate the process because we don't need it running at this point in the program.

There are various reasons for the GC-auto-terminator to not terminate the process, the GC being paused is one of them